source/common/common/token_bucket_impl.h

```diff
@@ -28,9 +28,10 @@ class TokenBucketImpl : public TokenBucket {
   void maybeReset(uint64_t num_tokens) override;
 
 private:
-  const double max_tokens_;
+  const uint64_t max_tokens_;
   const double fill_rate_;
-  double tokens_;
+  uint64_t tokens_;
+  uint64_t residual_tokens_{0};
```

Comment:
It would be good to add a comment for residual_tokens_ indicating that this holds "micro" tokens.  Changing the name to residual_micro_tokens_ may also aid with understanding.

Reply:
Thanks your CR!

Yes， The suggested name is better， I will update it。

Reply:
Thanks your CR!

Yes， The suggested name is better， I will update it。